### PR TITLE
catch error response from SlackOAuthProvider

### PR DESCRIPTION
**Pull request by camhashemi**
@leogrim looks like `SlackOAuthProvider[43]` is where we get this https://www.kifi.com/authenticate/slack?error=access_denied&state=8ff5913a-0638-4d4f-933f-fc9ffd3157ea from. we just need to handle those responses appropriately (for now going back to /login)

### CODE REVIEW

File: kifi-backend/modules/common/app/com/keepit/social/providers/SecureSocialControllers.scala [hunk 1]
```diff
@@ -80,7 +80,10 @@ object ProviderController extends Controller with Logging {
             user =>
               log.info(s"[handleAuth] user [${user.email} ${user.identityId}] found from provider - completing auth")
               completeAuthentication(user, request.session)
-          })
+          }) match {
+            case result if result.header.status == 400 => Redirect(RoutesHelper.login())
+            case result => result
+          }
         } catch {
           case ex: AccessDeniedException => {
```
Comment (camhashemi) [inline kifi-backend/modules/common/app/com/keepit/social/providers/SecureSocialControllers.scala:90]: seems like other OAuthProvider subclasses throw exceptions when things go bad, so we don't have ways of handling results with error codes.

Comment (leogrim) [inline kifi-backend/modules/common/app/com/keepit/social/providers/SecureSocialControllers.scala:90]: Weird

**Comment by leogrim** [comment]
:+1: we actually have a task to track that whole flow, could you add an airbrake there for now since you're at it?

**Comment by camhashemi** [comment]
np, the only case I know this breaks is when you press "cancel" instead of "authorize" on the slack auth page, so would be good to see what else this could catch.

this task? https://app.asana.com/0/32728208409723/86741480256014

**Comment by leogrim** [comment]
Nope not this task, `SlackOAuthProvider` only handles login and signup. That task is about all other Slack flows handled by `SlackOAuthController` I believe

**Comment by camhashemi** [comment]
updated, put airbrakes on the producer of the result and log.error on the consumer

**Pull request merged**